cpp/lib/Authenticator.cpp

```diff
@@ -363,5 +360,178 @@ namespace Client
 
     return coded;
   }
+
+  AuthenticatorOKTA::AuthenticatorOKTA(
+      SF_CONNECT* connection) : m_connection(connection)
+  {
+      m_account = m_connection->account;
+      m_authenticator = m_connection->authenticator;
+      m_user = m_connection->user;
+      m_password = m_connection->password;
+      m_port = m_connection->port;
+      m_host = m_connection->host;
+      m_protocol = m_connection->protocol;
+      m_disableSamlUrlCheck = m_connection->disable_saml_url_check;
+      m_retriedCount = 0;
+      m_retryTimeout = get_retry_timeout(m_connection);
+
+      //m_appID = m_connection->application_name;
+      //m_appVersion = m_connection->application_version;
+      m_appID = "ODBC";
```

Comment:
Is this feature needs server change and here is the temporary workaround to pass the test?
If so please comment in the ticket/PR and add TODO comment here to ensure this will be removed before release.

Reply:
Added the comment.